library/src/main/kotlin/com/freeletics/coredux/ReduxStore.kt

```diff
@@ -2,116 +2,121 @@ package com.freeletics.coredux
 
 import kotlinx.coroutines.CoroutineScope
 import kotlinx.coroutines.ExperimentalCoroutinesApi
-import kotlinx.coroutines.GlobalScope
-import kotlinx.coroutines.ObsoleteCoroutinesApi
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.cancel
 import kotlinx.coroutines.channels.BroadcastChannel
 import kotlinx.coroutines.channels.Channel
-import kotlinx.coroutines.channels.ReceiveChannel
-import kotlinx.coroutines.channels.toChannel
+import kotlinx.coroutines.channels.Channel.Factory.CONFLATED
+import kotlinx.coroutines.isActive
 import kotlinx.coroutines.launch
 
 /**
- * A ReduxStore is a RxJava based implementation of Redux and redux-observable.js.org.
- * A ReduxStore takes Actions from upstream as input events.
- * [SideEffect]s can be registered to listen for a certain
- * Action to react on a that Action as a (impure) side effect and create yet another Action as
- * output. Every Action goes through the a [Reducer], which is basically a pure function that takes
- * the current State and an Action to compute a new State.
- * The new state will be emitted downstream to any listener interested in it.
- *
- * A ReduxStore observable never reaches onComplete(). If a error occurs in the [Reducer] or in any
- * side effect ([Throwable] has been thrown) then the ReduxStore reaches onError() as well and
- * according to the reactive stream specs the store cannot recover the error state.
+ * A [reduxStore] is a Kotlin coroutine based implementation of Redux and redux.js.org.
  *
  * @param initialState The initial state. This one will be emitted directly in onSubscribe()
- * @param sideEffects The sideEffects. See [SideEffect]
+ * @param stateReceiver function that receives new state updates
+ * @param sideEffects The sideEffects. See [SideEffect].
  * @param reducer The reducer.  See [Reducer].
  * @param S The type of the State
  * @param A The type of the Actions
+ *
+ * @return [ActionDispatcher] that should be use to dispatch new actions to [reduxStore]
  */
-@UseExperimental(ExperimentalCoroutinesApi::class, ObsoleteCoroutinesApi::class)
-fun <S: Any, A: Any> ReceiveChannel<A>.reduxStore(
+@UseExperimental(ExperimentalCoroutinesApi::class)
+fun <S: Any, A: Any> CoroutineScope.reduxStore(
```

Comment:
Do think it makes sense to rename this function to `createStore()`? It creates a new redux store for you.

Reply:
No strong opinion either... `createStore()` makes sense to me ...

`reduxStore()` as a name might make more sense as an Rx operator